### PR TITLE
Fixing component handling for lcos

### PR DESCRIPTION
**Pull request by sithhell**
This patch fixes a UB where the LCOs using (simple) components where cast to
wrong types.

### CODE REVIEW

File: hpx/performance_counters/server/base_performance_counter.hpp [hunk 1]
```diff
@@ -19,153 +19,163 @@
...
-            start_nonvirt, start_action);
-
-        /// The \a stop_action
-        HPX_DEFINE_COMPONENT_ACTION(base_performance_counter,
-            stop_nonvirt, stop_action);
-
-    protected:
-        hpx::performance_counters::counter_info info_;
-        util::atomic_count invocation_count_;
+        class base_performance_counter
```
Comment (hkaiser) [inline hpx/performance_counters/server/base_performance_counter.hpp]: What happened here?

Comment (sithhell) [inline hpx/performance_counters/server/base_performance_counter.hpp]: This diff are reformating changes introduced by clang-format...

Comment (hkaiser) [inline hpx/performance_counters/server/base_performance_counter.hpp]: We agreed not to reformat whole files without dire need...

Comment (sithhell) [inline hpx/performance_counters/server/base_performance_counter.hpp]: I have no problem reverting the formatting changes again. I'm not exactly sure what happened here. I use `git clang-format` which should only affect the direct changeset (in that case, a forward declaration was removed). The resulting diff is pretty awful...

**Pull request merged**